### PR TITLE
fix!: remove unnecessary nonce from message gql queries

### LINKED ISSUE #3294: Optimize messages's graphql queries
During some load tests, we realized a couple of issues ([#3279](https://github.com/FuelLabs/fuels-ts/issues/3279), [#3277](https://github.com/FuelLabs/fuels-ts/issues/3277)) that we're fixing and optimizing.

In keeping with this theme, we should continue the effort and review our predefined GQL queries and fragments.

Every field removal _is_ a breaking change and _can_ break things for apps, so we need to be extra careful here.

Taking from the [`providers/operations.graphql`](https://github.com/FuelLabs/fuels-ts/blob/d0e9b1a110721fa721aa40f87d7b5515be16b9b3/packages/account/src/providers/operations.graphql):

- [x] `getMessageProof`
- [x] `getMessageStatus`
- [x] `getMessageByNonce`
- [x] `getMessages`

### PR DESCRIPTION
**Pull request by maschad**
- Closes #3294

# Release notes

In this release, we:

- Optimized the `getMessageByNonce` provider endpoint

# Summary

The `nonce` returned from the `gql` query was removed from the `messageFragment` operation as the parameter can actually be re-used in the function  `getMessageByNonce` and  a separate fragment was created for `getMessage`

# Breaking Changes

- Removed the `nonce` property from `Provider.operations.getMessageByNonce()`. This can still be retrieved by `Provider.getMessageByNonce()`.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: .changeset/chatty-camels-wave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@fuel-ts/account": patch
+---
+
+fix: remove unnecessary nonce from message gql queries
```
Comment (danielbate) [inline .changeset/chatty-camels-wave.md:5]:
```suggestion
---
"@fuel-ts/account": minor
---

fix!: remove unnecessary nonce from message gql queries
```

**Review by danielbate — APPROVED** [review]

File: packages/account/src/providers/operations.graphql [hunk 2]
```diff
@@ -260,10 +260,18 @@ fragment messageFragment on Message {
...
 
+fragment getMessageFragment on Message {
+  amount
+  sender
+  recipient
+  data
+  daHeight
+  nonce
+}
+
```
Comment (nedsalk) [inline packages/account/src/providers/operations.graphql:274]:
```suggestion
fragment getMessageFragment on Message {
  ...messageFragment
  nonce
}

```

**Review by nedsalk — APPROVED** [review]

**Comment by petertonysmith94** [comment]
This can be done at a later date - this optimization doesn't provide (much value).

**Comment by maschad** [comment]
Re-opening this so that we may re-introduce breaking changes at a later more appropriate date.

**Review by danielbate — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**